docs/runbook.md

```diff
@@ -13,103 +13,195 @@
+
 ## Reference
 
 ### name
 
+type: str, optional, default is "not_named"
+
+Set the name of the test run. This name will be used to group results and put it in title, also the created resources's name contains this specified str.
+
 ### test_project
 
+type: str, optional, default is empty
+
+Set the project name of this test run. This name will be used to group test results in html.
+
 ### test_pass
 
+type: str, optional, default is empty
+
+Set the test pass name of this test run. This name combined with test project will be used to group test results in html.
+
 ### tags
 
+type: list of str, optional, default is empty
+
+Set the tags of the test run. This name combined with test project and test pass names will be used to group test results in html.
+
 ### parent
 
+Share the runbook for similar runs.
+
 #### path
 
-#### strategy
+type: str, optional, default is empty
 
-Not implemented
+Specify the path of the parent runbook.
 
 ### extension
 
+type: list of str, optional, default: empty
+
+Specify the path of the LISA module for loading test cases and extensions.
+
 ### variable
 
+Used to support variables in other fields.
+
 #### is_secret
 
-#### file
+type: boolean, optional, default is False.
 
-#### name
+If the variable is secret, the value of this variable will be removed from log and other output information.
 
-### value
+#### file
 
-### artifact
+type: list of str, optional, default: empty
+
+Specify path of other yml files which define variables.
 
 #### name
 
-#### locations
+type: str, optional, default is empty.
 
-##### type
+Variable name.
 
-##### path
+#### value
+
+type: str, optional, default is empty
 
+Value of the paired variable.
 
 ### notifier
 
+Receive messages during the test run and output them somewhere. The supported messages are derived from `MessageBase`, including `TestRunMessage`, `TestResultMessage`, etc.
+
 #### console
 
+This is the simplest notifier. It outputs messages to the console and file log and demonstrates how to implement notification procedures.
+
+##### log_level
+
+type: str, optional, default: DEBUG, values: DEBUG, INFO, WARNING...
+
+Set log level of notification messages.
+
 #### html
 
+Output test results in html format. It can be used for local development or as the body of an email.
+
+##### path
+
+type: str, optional, default: lisa.html
+
+Specify the output file name and path.
+
+##### auto_open
+
+type: bool, optional, default: False
+
+When set to True, the html will be opened in the browser after completion. Useful in local run.
 
 ### environment
 
-#### name
+List of environments.
+
+#### environments
+
+Define test run environment.
+
+##### name
 
-#### topology
+type: str, optional, default is empty
 
-#### nodes
+Specify the name of the environment.
 
-#### nodes_requirement
+##### topology
 
-##### type
+type: str, optional, default is "subnet"
+
+Specify the topology of the environment.
+
+##### nodes
+
+List of system under test.
```

Comment:
Just call node, not take any other names. You can introduce node is virtual machines in most testing, may be barelmetal or others.